src/fios.cpp

```diff
@@ -205,12 +205,16 @@ const char *FiosBrowseTo(const FiosItem *item)
  */
 static void FiosMakeFilename(char *buf, const char *path, const char *name, const char *ext, const char *last)
 {
-	const char *period;
+	assert(*path != '\0' && *name != '\0');
+
+	buf = strecpy(buf, path, last);
+	/* Remove trailing path separator, if present */
+	if (buf[-1] == PATHSEPCHAR) buf--;
 
 	/* Don't append the extension if it is already there */
-	period = strrchr(name, '.');
+	const char *period = strrchr(name, '.');
 	if (period != nullptr && strcasecmp(period, ext) == 0) ext = "";
-	seprintf(buf, last, "%s" PATHSEP "%s%s", path, name, ext);
+	seprintf(buf, last, PATHSEP "%s%s", name, ext);
```

Comment:
perhaps something more like

```suggestion
	seprintf(buf, last, "%s%s%s%s", path, path[strlen(path) - 1] != PATHSEPCHAR ? PATHSEP : "", name, ext);
```

Reply:
> **path[strlen(path) - 1]** != PATHSEPCHAR
This also (as my previous code) could lead to an wrong result, because we are dealing with UTF-8.